manifests/init.pp

```diff
@@ -4,6 +4,8 @@
 #
 class hosts (
   $collect_all           = false,
+  $collect_tag           = undef,
```

Comment:
spec tests are needed for the new params. They will need to show that the wrong types will fail.